prow/sidecar/censor.go

```diff
@@ -418,10 +419,79 @@ func loadSecrets(paths []string) ([][]byte, error) {
 				return err
 			}
 			secrets = append(secrets, raw)
+			var parser = func(bytes []byte) ([]string, error) {
+				return []string{}, nil
+			}
+			if info.Name() == ".dockercfg" {
+				parser = loadDockercfgAuths
+			}
+			if info.Name() == ".dockerconfigjson" {
+				parser = loadDockerconfigJsonAuths
+			}
+			extra, parseErr := parser(raw)
+			if parseErr != nil {
+				return fmt.Errorf("could not read %s as a docker secret: %v", path, parseErr)
+			}
+			// It is important that these are added to the list of secrets *after* their parent data
+			// as we will censor in order and this will give a reasonable guarantee that the parent
+			// data (a superset of any of these fields) will be censored in its entirety, first. It
+			// remains possible that the sliding window used to censor pulls in only part of the
+			// superset and some small part of it is censored first, making the larger superset no
+			// longer match the file being censored.
```

Comment:
We could double-parse the file if we *really* wanted to enforce this, or we could make this behavior opt-in. However, I think it's probably best as-is ... if we hit this case, we will still censor all of the "sensitive" parts of the larger secret and users have and will continue to leak just the important part of a pull secret, not the whole thing in all of it's JSON glory.